src/bridge/cockpitdbusjson.c

```diff
@@ -788,25 +788,34 @@ static JsonObject *
 build_json_error (GError *error)
 {
   JsonObject *object;
-  JsonArray *reply;
-  JsonArray *args;
-  gchar *error_name;
 
   object = json_object_new ();
-  reply = json_array_new ();
-  args = json_array_new ();
 
-  error_name = g_dbus_error_get_remote_error (error);
-  g_dbus_error_strip_remote_error (error);
+  if (g_error_matches(error, G_IO_ERROR, G_IO_ERROR_TIMED_OUT))
+    {
+      json_object_set_string_member (object, "error", "timeout");
```

Comment:
I think it's worth faking a DBus error here. We do the same thing with ```InvalidArgs```. We pretend to have received that from the service, even though we short-circuited and didn't even send a message call. This normalizes the error handling in the javascript ... and ended up solving one of the big issues that Simon McVittie regretted in dbus-glib.

Perhaps ```org.freedesktop.DBus.Error.NoReply``` or ```org.freedesktop.DBus.Error.Timeout```. Although the later seems more intuitive ... I believe the former is used regularly (eg: in the dbus-daemon). Worth checking.